packages/material-ui/src/Menu/Menu.js

```diff
@@ -34,6 +34,14 @@ export const styles = {
     // We disable the focus ring for mouse, touch and keyboard users.
     outline: 0,
   },
+  /* Styles applied to sub`Menu`s to prevent the Modal from capturing mouse events */
+  disablePointerEvents: {
```

Comment:
For one-off styles we can use the existing styling engine. Adding a prop for each kind of pointer-event is a bit excessive.

Reply:
I'm a little confused by 
>Adding a prop for each kind of pointer-event is a bit excessive.

I assume I'm missing something, but, don't these style classes (not props?) already make use of the existing styling engine? Maybe I just don't know how to refer to/think about them correctly.

A bit more on why these are here: This is how I got around the click away mask that `Menu`s have (coming from the `Popover`) for all `subMenu`s, so you can directly interact with any visible layer of the cascading menu even though all `subMenu`s make use of the same `Menu` component as the root menu. So these two classes (`enablePointerEvents` and `disablePointerEvents`) are conditionally assigned using `clsx` based on the `isSubMenu` prop value.

Maybe there's a better way to do this? It seems to work well in all my testing, but it's quite possible there's a gotcha I don't know about.

Reply:
I meant: Disable pointer-events at the callsite of the component and don't add a public API for it. Now that I think about it: I wouldn't recommend pointer-events: none at all since it also disables hover styles and tooltips. 

Could you describe the classname at a higher level i.e. what are you trying to achieve? Right now the classname is just saying how it works but not what or why it should be used.

Reply:
This is a longer answer and it's way past quitting time, at this point. I'll follow up tomorrow with a better explanation of what I'm doing by disabling and re-enabling pointer-events on different layers of the `Popover`. I'd honestly love to find a better, less hacky way of accomplishing what I'm after.

Reply:
Alright. Let me try to speak to this usefully.

Basically, there are behaviors coming from the `Modal` that a `Menu` renders which prevent clicking on parts of the UI which are not part of the currently visible `Menu`. Under normal circumstances this is fine, but for a Cascading Menu where we have multiple `Menu` components visible at once, and where we want to be able to interact with all visible parts of that Cascading Menu, it becomes a problem.

In its normal state, you can only interact with the deepest level of the cascading menu at any given moment unless you modify the behavior.

I tried to find a way to control the `TrapFocus` and `BackdropComponent` parts of the `Modal` from the `Menu` api, but couldn't find a way that worked.

What ultimately ended up working was to actually disable all mouse events on the `Popover` itself, which includes the problematic `Modal`, and then re-enable pointer events on the `Paper` that the `Popover` renders its children in, effectively bypassing the entire problematic layer for all `SubMenu`s, while leaving the useful behavior alone on the root `Menu` node.

I'd love to find a better way, but (to be transparent) another goal early on was to minimize the number of modified files so it seemed less impactful and more likely to gain traction from the community and maintainers, so I didn't want to go modifying other API's just to support this weird, hard-to-explain requirement. =P

Sorry for the novel. Hopefully that helps describe the thinking and some of the history for those two imperative mouse event styles.